src/main/java/org/prebid/server/analytics/AnalyticsReporterDelegator.java

```diff
@@ -72,4 +90,79 @@ public AnalyticsReporterDelegator(List<AnalyticsReporter> delegates,
                     tcfContext.getConsentString(), delegates);
         }
     }
+
+    private <T> void validateEvent(T event) {
+        if (event instanceof AuctionEvent) {
+            validateExtPrebidAnalytics((AuctionEvent) event);
+        }
+    }
+
+    private void validateExtPrebidAnalytics(AuctionEvent auctionEvent) {
+        final ExtRequest requestExt = auctionEvent.getAuctionContext().getBidRequest().getExt();
+        final ExtRequestPrebid extPrebid = requestExt != null ? requestExt.getPrebid() : null;
+        final ObjectNode analytics = extPrebid != null ? extPrebid.getAnalytics() : null;
+        final Iterator<String> analyticsFieldNames = analytics != null && !analytics.isEmpty()
+                ? analytics.fieldNames() : null;
+
+        if (analyticsFieldNames != null) {
+            final List<String> unknownAdapterNames = StreamSupport.stream(
+                    Spliterators.spliteratorUnknownSize(analyticsFieldNames, Spliterator.ORDERED), false)
+                    .filter(adapter -> !reporterNames.contains(adapter))
+                    .collect(Collectors.toList());
+            if (CollectionUtils.isNotEmpty(unknownAdapterNames)) {
+                logger.warn(
+                        String.format("Unknown adapters in ext.prebid.analytics[].adapter: %s", analyticsFieldNames));
+            }
+        }
+    }
+
+    private static <T> T updateEvent(T event, String delegatorAdapter) {
+        if (!ADAPTERS_PERMITTED_FOR_FULL_DATA.contains(delegatorAdapter) && event instanceof AuctionEvent) {
+            final AuctionEvent auctionEvent = (AuctionEvent) event;
+            final AuctionContext updatedAuctionContext =
+                    updateAuctionContextForDelegator(auctionEvent.getAuctionContext(), delegatorAdapter);
+            return updatedAuctionContext != null
+                    ? (T) auctionEvent.toBuilder().auctionContext(updatedAuctionContext).build()
+                    : event;
+        }
+
+        return event;
+    }
+
+    private static AuctionContext updateAuctionContextForDelegator(AuctionContext context, String delegatorAdapter) {
```

Comment:
1. Rename to `updateAuctionContextAdapter`.
2. Rename `delegatorAdapter` to `adapter`.